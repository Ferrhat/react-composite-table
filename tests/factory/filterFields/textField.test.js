import React from 'react';
import {shallow, mount} from 'enzyme';
import TextField from "../../../lib/factory/filterFields/textField";

describe('TextField', () => {
    const mockOnChange = jest.fn();
    const textField = shallow(<TextField onChange={mockOnChange} filterableProperty={'testValue'} name={'testName'} />);

    it('renders properly', () => {
        expect(textField).toMatchSnapshot();
    });

    it('contains an input component', () => {
        expect(textField.find('input').length).toEqual(1);
    });

    it('stores the changed value', () => {
        textField.setProps({value: 'newValue'});
        expect(textField.state('value')).toEqual('newValue');
    });

    it('should trigger a change event', () => {
        const event = {
            preventDefault() {},
            target: { name: 'testName', value: 'testValue' }
        };
        textField.find('input').simulate('change', event);
        expect(mockOnChange).toBeCalledWith('testName', 'text', 'testValue', event.target.value);
        expect(textField.state('value')).toEqual(event.target.value);
    });

});
