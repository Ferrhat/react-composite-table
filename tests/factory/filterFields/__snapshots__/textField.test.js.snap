// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField renders properly 1`] = `
<input
  className="n-inputfield form-control"
  onChange={[Function]}
  style={
    Object {
      "height": "100%",
    }
  }
  type="text"
  value=""
/>
`;
